sound/soc/sof/ipc.c

```diff
@@ -1015,8 +760,9 @@ struct snd_sof_ipc *snd_sof_ipc_init(struct snd_sof_dev *sdev)
 	ipc->ops = &ipc3_ops;
 
 	/* check for mandatory ops */
-	if (!ipc->ops->get_reply || !ipc->ops->pcm || !ipc->ops->tplg ||
-	    !ipc->ops->tplg->widget || !ipc->ops->tplg->control) {
+	if (!ipc->ops->get_reply || !ipc->ops->tx_msg || !ipc->ops->set_get_data ||
+	    !ipc->ops->pcm || !ipc->ops->tplg || !ipc->ops->tplg->widget ||
```

Comment:
ah, that answers partially to my previous comment but I still don't see rx_msg?

Reply:
Right, now I got it.

First I have switched to get_reply ops, with that change the given ops is made mandatory.
This patch switches to the tx_msg and set_get_data ops, so from this point these are mandatory.

The rx_msg still has full implementation in ipc.c, so it is not mandatory, but if I'm not mistaken when we switch to the rx_msg ops, then it is also going to be mandatory.